05_casts/05_00_Interesting_cases/05_00_02_reinterpret_cast.cpp

```diff
@@ -0,0 +1,25 @@
+// Show that with reinterpret_cast you need to understand what you are doing,
+// the last case is possible.
+
+#include <iostream>
+
+struct foo {
+    void my_foo() {
+        std::cout << "foo" << std::endl;
+    }
+};
+
+struct bar {
+    void my_bar() {
+        std::cout << "bar" << std::endl;
+    }
+};
+
+int main() {
+    foo* foo_ptr = new foo();
```

Comment:
memory leak. I don't have strong feeling we need dynamic memory is any of the examples. Do it really help?